tests/acceptance/blocking_adapter_test.py

```diff
@@ -2393,5 +2401,23 @@ def test(self):
 ##        self.assertEqual(frame.method.consumer_count, 1)
 
 
+class TestConnectionAttemptsAgainstTimeout(BlockingTestCaseBase):
+
+    def test_connection_attempts_1(self):
+        params = pika.ConnectionParameters(connection_attempts=1)
+        with mock.patch('socket.socket.connect', side_effect=socket.timeout()):
```

Comment:
@gst, we do unit/mock tests in the `tests/unit/` directory. The acceptance tests are performed with real connections or connection attempts.  We already had a similar acceptance test `TestConnectWithDownedBroker`. I extended it to support 1 and 2 connection attempts: https://github.com/vitaly-krugl/pika/commit/81cbba61f4d8db1fd6c1db1650fbfe3acb0390df